examples/src/main/cc/wfa/examples/streaming/hello_streaming_server.cc

```diff
@@ -1,70 +1,68 @@
-// limitations under the License.
+/*
+ * Copyright 2021 The Cross-Media Measurement Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
+#include <grpc/grpc.h>
 #include <grpcpp/server.h>
 #include <grpcpp/server_builder.h>
+#include <grpcpp/server_context.h>
+#include <grpcpp/security/server_credentials.h>
+#include "src/main/proto/wfa/examples/streaming/hello_streaming_service.grpc.pb.h"
 
-#include <iostream>
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerReader;
+using grpc::ServerContext;
+using grpc::Status;
+using wfa_examples::streaming::HelloStreaming;
+using wfa_examples::streaming::SayHelloStreamingRequest;
+using wfa_examples::streaming::SayHelloStreamingResponse;
 
-#include "absl/strings/str_cat.h"
-#include "src/main/cc/wfa/examples/streaming/hello_streaming_service.h"
-
-namespace wfa_examples {
-namespace streaming {
-namespace {
-
-using ::grpc::Server;
-using ::grpc::ServerBuilder;
-
-void RunServer(int port) {
-  HelloStreamingService service;
-  ServerBuilder builder;
-  builder.RegisterService(&service);
-
-  builder.AddListeningPort(absl::StrCat("[::]:", port),
-                           grpc::InsecureServerCredentials());
-  std::unique_ptr<Server> server = builder.BuildAndStart();
-  std::cerr << "Server listening on port " << port << std::endl;
-  server->Wait();
-}
-
-}  // namespace
-}  // namespace streaming
-}  // namespace wfa_examples
-
-namespace {
-
-constexpr int kDefaultPort = 50051;
-
-int GetPort(int argc, char** argv) {
-  if (argc > 1) {
-    int port;
-    if (absl::SimpleAtoi(argv[1], &port)) {
-      return port;
-    } else {
-      std::cerr << "Failed to parse port from " << argv[1] << std::endl;
+class HelloStreamingImpl final : public HelloStreaming::Service {
+ public:
+  Status SayHelloStreaming(ServerContext* context,
+                           ServerReader<SayHelloStreamingRequest>* reader,
+                           SayHelloStreamingResponse* response) override {
+    SayHelloStreamingRequest request;
+    while (reader->Read(&request)) {
+      std::cout << "Received: " << request.name() << std::endl;
+      response->add_message("Hello " + request.name());
     }
+
+    return Status::OK;
   }
+};
 
-  return kDefaultPort;
+std::string GetServerAddress() {
+  std::string server_host("0.0.0.0");
+  std::string server_port("50051");
+  std::string server_address = server_host + ":" + server_port;
+  return server_address;
```

Comment:
This seems unnecessary, maybe just write `std::string server_address = "0.0.0.0:50051";` in main.